### ISSUE TITLE
findJsonPaths infinite recursion

### ISSUE THREAD

**Issue by mgroth0** [cbeust/klaxon#132]
Annotations.findJsonPaths is doing an infinite recursion and blocking the thread. I will post more details once I figure it out.

**Comment by mgroth0** [comment]
Maybe its because my class was javafx.scene.node which contains properties of type Node?

**Comment by mgroth0** [comment]
Maybe an exception can be thrown in cases like this to save time debugging.

**Comment by cbeust** [comment]
That's quite possible, can you post a snippet reproducing the problem?

**Comment by mgroth0** [comment]
```kotlin
fun main(args: Array<String>) {
    Application.launch(KlaxonRecursion::class.java)
}

class KlaxonRecursion : Application() {
    override fun start(primaryStage: Stage?) {
        val converter = object : Converter<Node> {
            override fun toJson(value: Node): String? {
                return "string"
            }

            override fun fromJson(jv: JsonValue): Node {
                return Label()
            }
        }
    Klaxon().converter(converter).parse<Node>("blah blah")
    }
}
```

**Comment by cbeust** [comment]
Fixed in 3.0.1.
